controllers/users.js

```diff
@@ -1,37 +1,152 @@
-const jwt = require("jsonwebtoken");
+const jwt = require('jsonwebtoken');
 
 function userDefinedException(message, statusCode) {
   this.message = message;
   this.statusCode = statusCode;
 }
+function getRandomInt(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+function isStoredTimeWithin10Minutes(otpGeneratedAt) {
+  // Convert stored time to Date object
+  const storedDate = new Date(otpGeneratedAt);
+
+  // Get current time
+  const currentDate = new Date();
+
+  // Calculate difference in milliseconds
+  const differenceInMilliseconds = currentDate - storedDate;
+
+  // Convert difference to minutes
+  const differenceInMinutes = differenceInMilliseconds / (1000 * 60); // 1000 milliseconds in a second, 60 seconds in a minute
+
+  // Check if difference is less than 10 minutes
+  return differenceInMinutes < 10;
+}
+
 function auth(app, sequelize) {
   const { User } = sequelize;
-  app.post("/login", async function login(req, res) {
-    const { username, password } = req.body;
+  app.post('/sendOtp', async function sendOtp(req, res) {
+    const { mobileNumber } = req.body;
     try {
-      if (!username || !password) {
-        throw new userDefinedException(
-          "Please enter both Email and Password",
-          401
-        );
+      if (!mobileNumber) {
+        throw new userDefinedException('Please enter mobile number', 401);
       }
-      const user = await User.findOne({ where: { username } });
-      console.log(user);
+      const user = await User.findOne({ where: { mobileNumber } });
       if (!user) {
-        throw new userDefinedException("No user found!", 404);
+        throw new userDefinedException('No user found!', 404);
       }
-      if (password !== user.password) {
-        throw new userDefinedException("Wrong password!", 401);
+
+      if (!user.otp) {
+        const randomInt = getRandomInt(1000, 9999); // Generates a random integer between 1 and 100
+        console.log(randomInt);
+        const otpGeneratedAt = new Date();
+        let retryCount = user.retryCount + 1;
+        //add send Interakt notification function
+        const updatedUser = await user.update({
+          otp,
+          retryCount,
+          otpGeneratedAt
+        });
+        console.log(updatedUser);
+        return res
+          .status(200)
+          .json({ message: `Otp sent successfully to ${mobileNumber}` });
+      } else {
+        let isOtpValid;
+        isOtpValid = isStoredTimeWithin10Minutes(user.otpGeneratedAt);
+        if (user.retryCount < 3) {
```

Comment:
use a constant variable instead of the number 3. Something like MAX_RETRY_COUNT 
Also, it should be less than 4. Since in your first attempt, you are setting it to 1, and then to 2, it would already be 3 in your third attempt. Hence never making it to the this condition. 

Use the early return principle here. combine user.retryCount < 3 and isOtpValid in a single if